fish-rust/Cargo.toml

```diff
@@ -29,6 +29,7 @@ rand = { version = "0.8.5", features = ["small_rng"] }
 unixstring = "0.2.7"
 widestring = "1.0.2"
 rand_pcg = "0.3.1"
+parking_lot = "0.12.1"
```

Comment:
parking_lot is a drop-in replacement for the standard sync primitives?
We should not have multiple ways of doing the same thing without a good reason.

Reply:
The `std::sync::RwLock` does not support upgradeable readers/downgradeable writer, which is necessary for `scoped_test` to atomically change from a writer to a reader for the test to actually read the feature value. Sadly the diff for adding `scoped_test` got mangled. This is getting added to avoid race-conditions as tests in Rust are multi-threaded by default, unlike the C++ ones,  see https://github.com/fish-shell/fish-shell/pull/9854#discussion_r1257339267 for where this will be used.

We should be able to use `parking_lot` as a dev-dependency and use `std::sync:RwLock` otherwise, but I don't see the point in doing so as it just causes unnecessary `cfg[]`'s ,and `parking_lot::RwLock` is generally faster afaik https://crates.io/crates/parking_lot

Reply:
can we tell Rust to not run this test in parallel to others?
In Go you have to explicitly enable parallelism for tests..

Reply:
That requires either `cargo test -- --test-threads=1` or using a separate test-runner like https://nexte.st and using test groups https://nexte.st/book/test-groups.html